src/client/components/PageLayout/Header/CycleSwitcher/hooks/useNavigateTo.tsx

```diff
@@ -21,13 +24,14 @@ export const useNavigateTo = () => {
       const hasRoleInCountry = Users.hasRoleInCountry({ user, cycle, countryIso })
 
       let link = ''
-      if (countryIso && (cycle.published || hasRoleInCountry))
+      if (isTutorialsRoute) link = Routes.Cycle.generatePath({ assessmentName, cycleName })
```

Comment:
Okay! By generic rule do you mean the code here or in the routing system?
In this line specifically, this would be a especial case like `else if (isAdminPage)`3 lines below.